src/test/msixtest/unbundle.cpp

```diff
@@ -105,11 +105,122 @@ TEST_CASE("Unbundle_StoreSigned_Desktop_x86_x64_MoviesTV_lang_applicability_off"
     allFiles.insert(filesNotApplicable.begin(), filesNotApplicable.end());
+
+    CHECK(MsixTest::Directory::CompareDirectory(outputDir, allFiles));
+
+    // Clean directory
+    CHECK(MsixTest::Directory::CleanDirectory(outputDir));
+}
+
+TEST_CASE("Unbundle_StoreSigned_Desktop_x86_x64_MoviesTV_pfn_extract-all", "[unbundle]")
+{
+    HRESULT expected = S_OK;
+    std::string bundle = "StoreSigned_Desktop_x86_x64_MoviesTV.appxbundle";
+    MSIX_VALIDATION_OPTION validation = MSIX_VALIDATION_OPTION_FULL;
+    MSIX_PACKUNPACK_OPTION packUnpack = MSIX_PACKUNPACK_OPTION_CREATEPACKAGESUBFOLDER;
+    MSIX_APPLICABILITY_OPTIONS applicability = static_cast<MSIX_APPLICABILITY_OPTIONS>(MSIX_APPLICABILITY_NONE);
+
+    RunUnbundleTest(expected, bundle, validation, packUnpack, applicability, MsixTest::TestPath::Directory::Unbundle, false);
+
+    // The expected folder structure should be <output>/Microsoft.ZuneVideo_2019.6.25071.0_neutral_~_8wekyb3d8bbwe/<files>
+    // Append it to the already existing expected files map
+    std::string pfn = "Microsoft.ZuneVideo_2019.6.25071.0_neutral_~_8wekyb3d8bbwe/";
+    auto files = MsixTest::Unbundle::GetExpectedFilesFullApplicable();
+    std::map<std::string, uint64_t> filesWithPfn;
+    for (const auto& file : files)
+    {
+        filesWithPfn.emplace(pfn + file.first, file.second);
+    }
+
+    auto resourceFiles = MsixTest::Unbundle::GetExpectedFilesNoApplicable();
+    for (const auto& file : resourceFiles)
+    {
+        filesWithPfn.emplace(pfn + file.first, file.second);
+    }
+
+    auto outputDir = MsixTest::TestPath::GetInstance()->GetPath(MsixTest::TestPath::Directory::Output);
+    CHECK(MsixTest::Directory::CompareDirectory(outputDir, filesWithPfn));
+
+    // Clean directory
+    CHECK(MsixTest::Directory::CleanDirectory(outputDir));
+}
+
+TEST_CASE("Unbundle_StoreSigned_Desktop_x86_x64_MoviesTV_pfn-flat", "[unbundle]")
+{
+    HRESULT expected = S_OK;
+    std::string bundle = "StoreSigned_Desktop_x86_x64_MoviesTV.appxbundle";
+    MSIX_VALIDATION_OPTION validation = MSIX_VALIDATION_OPTION_FULL;
+    MSIX_PACKUNPACK_OPTION packUnpack = MSIX_PACKUNPACK_OPTION_UNPACKWITHFLATSTRUCTURE;
+    MSIX_APPLICABILITY_OPTIONS applicability = MSIX_APPLICABILITY_OPTION_FULL;
+
+    RunUnbundleTest(expected, bundle, validation, packUnpack, applicability, MsixTest::TestPath::Directory::Unbundle, false);
+
+    auto outputDir = MsixTest::TestPath::GetInstance()->GetPath(MsixTest::TestPath::Directory::Output);
+
+    // Verify all the files extracted on disk are correct
+    auto files = MsixTest::Unbundle::GetExpectedBundleFilesPfnFlatFullApplicable();
+    CHECK(MsixTest::Directory::CompareDirectory(outputDir, files));
+
+    std::cout << "Expected Files Present" << '\n';
+
+    // Verify non expected files aren't present
+    auto filesNotApplicable = MsixTest::Unbundle::GetExpectedFilesNoApplicable();
+    CHECK(!MsixTest::Directory::CompareDirectory(outputDir, filesNotApplicable));

```

Comment:
>!MsixTest::Directory::CompareDirectory [](start = 10, length = 38)

This is a dubious check; but I know it is used other places in this file.  In reality, we would want CompareDirectory to return an enum that indicates {no files, some files, all files} and then we can make better judgements here.

Reply:
Yeah actually this entire check is completely busted and is  totally my fault. This is always going succeed if the fist file in map doesn't exists in the directory. 
If you want to fix it, you can go with what John suggested? Let me know if you need help with it 

---
In reply to: [289545863](https://github.com/Microsoft/msix-packaging/pull/150#discussion_r289545863) [](ancestors = 289545863)

Reply:
Actually, just get rid of the entire check, as the first call verifies that the exact map of files is present, so verifying that some other set of files isn't present is not needed.  Can also delete the other instance of using !CompareDirectory.